haystack/preview/components/generators/cohere/cohere.py

```diff
@@ -0,0 +1,132 @@
+import logging
+import sys
+from typing import Any, Callable, Dict, List, Optional
+
+from haystack.lazy_imports import LazyImport
+from haystack.preview import DeserializationError, component, default_from_dict, default_to_dict
+
+with LazyImport(message="Run 'pip install cohere'") as cohere_import:
+    from cohere import Client
+
+logger = logging.getLogger(__name__)
+
+
+API_BASE_URL = "https://api.cohere.ai"
+
+
+def default_streaming_callback(chunk):
+    """
+    Default callback function for streaming responses from Cohere API.
+    Prints the tokens of the first completion to stdout as soon as they are received and returns the chunk unchanged.
+    """
+    print(chunk.text, flush=True, end="")
+
+
+@component
+class CohereGenerator:
+    """LLM Generator compatible with Cohere's generate endpoint
+
+    Queries the LLM using Cohere's API. Invocations are made using 'cohere' package
+    See [Cohere API](https://docs.cohere.com/reference/generate) for more details.
+    """
+
+    def __init__(
+        self,
+        api_key: str,
+        model: str = "command",
+        streaming_callback: Optional[Callable] = None,
+        api_base_url: str = API_BASE_URL,
+        **kwargs,
+    ):
+        """
+        Args:
+            api_key (str): The API key for the Cohere API.
+            model_name (str): The name of the model to use.
+            streaming_callback (Callable, optional): A callback function to be called with the streaming response. Defaults to None.
+            api_base_url (str, optional): The base URL of the Cohere API. Defaults to "https://api.cohere.ai".
```

Comment:
We use a different style for docstrings: it's quite important for our automated API documentation tooling, so let's fix it.

Here is how it should look like:

```suggestion
        Instantiates a `CohereGenerator` component.

        :param api_key: The API key for the Cohere API.
        :param model_name: The name of the model to use.
        :param streaming_callback: A callback function to be called with the streaming response. Defaults to None.
        :param api_base_url: The base URL of the Cohere API. Defaults to "https://api.cohere.ai".
        :param kwargs: additional model parameters. These will be used during generation.
```

In addition:
- for model_name, let's give some example model names.
- for kwargs, let's add a link to the Cohere documentation where these arguments are listed.

Reply:
I was thinking the same, should have stuck with the way it is in GptGenerator. For the Kwargs, I have added the params in doc_string.